### PR TITLE
KAFKA-8591: WorkerConfigTransformer NPE on connector configuration reloading

### PR DESCRIPTION
**Pull request by nachomdo**
A bug in `WorkerConfigTransformer` prevents the connector configuration reload when the ConfigData TTL expires. 

The issue boils down to the fact that `worker.herder().restartConnector` is receiving a null callback. 

```
[2019-06-17 14:34:12,320] INFO Scheduling a restart of connector workshop-incremental in 60000 ms (org.apache.kafka.connect.runtime.WorkerConfigTransformer:88)
[2019-06-17 14:34:12,321] ERROR Uncaught exception in herder work thread, exiting:  (org.apache.kafka.connect.runtime.distributed.DistributedHerder:227)
java.lang.NullPointerException
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder$19.onCompletion(DistributedHerder.java:1187)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder$19.onCompletion(DistributedHerder.java:1183)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.tick(DistributedHerder.java:273)
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.run(DistributedHerder.java:219)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```
This patch keeps the same behaviour than before in the `WorkerConfigTransformer` in terms of ignoring any error returned from the callback. Do we still want to behave in the same way or we would like to handle any potential error? 🤔 maybe we can rely on the `scheduleReload` retrying again whenever the TTL expires or the connector task fails due to a stale configuration. 

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfigTransformer.java [hunk 1]
```diff
@@ -86,7 +87,8 @@ private void scheduleReload(String connectorName, String path, long ttl) {
             }
         }
         log.info("Scheduling a restart of connector {} in {} ms", connectorName, ttl);
-        HerderRequest request = worker.herder().restartConnector(ttl, connectorName, null);
+        FutureCallback<Void> cb = new FutureCallback<>();
```
Comment (rayokota) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfigTransformer.java]: Perhaps instead

```
CallBack<Void> cb = new Callback<Void>() {
    @Override
    public void onCompletion(Throwable error, Void result) {
        if (error != null) {
            log.error("Unexpected error during connector restart: ", error);
        }
    }
};
```

**Review by rayokota — COMMENTED** [review]
Thanks for the PR @nachomdo !  Left a small comment.

**Comment by nachomdo** [comment]
Feedback applied ⚡️ Thanks for the  👀 @rayokota! 🏆

**Review by rayokota — APPROVED** [review]
Thanks @nachomdo !  LGTM

@rhauch, can you review and merge when you get a chance?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java [hunk 2]
```diff
@@ -95,13 +96,13 @@ public void testReplaceVariableWithTTLAndScheduleRestart() {
     @Test
     public void testReplaceVariableWithTTLFirstCancelThenScheduleRestart() {
         EasyMock.expect(worker.herder()).andReturn(herder);
-        EasyMock.expect(herder.restartConnector(1L, MY_CONNECTOR, null)).andReturn(requestId);
+        EasyMock.expect(herder.restartConnector(eq(1L), eq(MY_CONNECTOR),  anyObject())).andReturn(requestId);
```
Comment (hachikuji) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: nit: extra space before `anyObject`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java [hunk 3]
```diff
@@ -84,8 +86,7 @@ public void testReplaceVariableWithTTL() {
     @Test
     public void testReplaceVariableWithTTLAndScheduleRestart() {
         EasyMock.expect(worker.herder()).andReturn(herder);
-        EasyMock.expect(herder.restartConnector(1L, MY_CONNECTOR, null)).andReturn(requestId);
-
+        EasyMock.expect(herder.restartConnector(eq(1L), eq(MY_CONNECTOR), anyObject())).andReturn(requestId);
```
Comment (hachikuji) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: Hmm.. These tests all pass even if the callback is null. Perhaps we can use `EasyMock.notNull()`?

**Review by hachikuji — COMMENTED** [review]
Thanks for the fix. Left a couple small comments.

**Comment by hachikuji** [comment]
retest this please

**Review by hachikuji — APPROVED** [review]
Thanks for the patch!

**Pull request merged**